### PR TITLE
Add app.kubernetes.io/version label

### LINKED ISSUE #5369: Include app.kubernetes.io/version as part of the recommended lables
If I create a new chart with `helm create my-chart` the following labels are included:
```
    app.kubernetes.io/name: {{ include "my-chart.name" . }}
    helm.sh/chart: {{ include "my-chart.chart" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
```

I think that `app.kubernetes.io/version: {{ .Chart.AppVersion }}` should also be included.

### PR DESCRIPTION
**Pull request by luisdavim**
**What this PR does / why we need it**:
`app.kubernetes.io/version` is a recommended label and I don't see any reason not to include it.

**Special notes for your reviewer**:
fixes #5369

### CODE REVIEW

**Comment by bacongobbler** [comment]
cc @mattfarina

**Comment by luisdavim** [comment]
Any updates on this?

**Comment by luisdavim** [comment]
I was waiting for this to be merged to open another PR to address #5372 but since it's taking a while to get this one through, I figured it would be better to address both issues in one go. So I've updated the PR, please let me know what you think.

File: cmd/helm/testdata/testcharts/alpine/templates/alpine-pod.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ metadata:
     # The "release" convention makes it easy to tie a release to all of the
     # Kubernetes resources that were created as part of that release.
     app.kubernetes.io/instance: {{.Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (mattfarina) [inline cmd/helm/testdata/testcharts/alpine/templates/alpine-pod.yaml]: The reason this wasn't included for these test data before was that the Charts themselves don't define an `AppVersion` so this is going to set it to an empty string. It's ok not to include something if there will be no value. This thought applies to all the label changes on this particular testdata chart.

File: docs/chart_template_guide/variables.md [hunk 2]
```diff
@@ -114,6 +114,7 @@ metadata:
     # I cannot reference .Chart.Name, but I can do $.Chart.Name
     helm.sh/chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
     app.kubernetes.io/instance: "{{ $.Release.Name }}"
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (mattfarina) [inline docs/chart_template_guide/variables.md]: Can you please add a comment above this to advice people to use the `app.kubernetes.io/version` if there is an `AppVersion` in their `Chart.yaml` file.

File: docs/examples/nginx/templates/configmap.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
     app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (mattfarina) [inline docs/examples/nginx/templates/configmap.yaml]: Before using `AppVersion` here can you add it to the `Chart.yaml` so there will be a value.

File: pkg/chartutil/create.go [hunk 4]
```diff
@@ -297,17 +288,25 @@ Create chart name and version as used by the chart label.
...
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "<CHARTNAME>.labels" -}}
+app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
+helm.sh/chart: {{ include "<CHARTNAME>.chart" . }}
+app.kubernetes.io/instance: {<CHARTNAME>se.Name }}
```
Comment (mattfarina) [inline pkg/chartutil/create.go]: I think this is supposed to be `.Release.Name`

File: pkg/lint/rules/testdata/albatross/templates/svc.yaml [hunk 5]
```diff
@@ -7,6 +7,7 @@ metadata:
   labels:
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
     app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (mattfarina) [inline pkg/lint/rules/testdata/albatross/templates/svc.yaml]: The `Chart.yaml` should include a version if it's going to be called here. We should either omit the `AppVersion` or add it to the `Chart.yaml`

**Review by mattfarina — COMMENTED** [review]
Thanks for PR. I'm happy to see better labeling in use. These are just some clarification nits.

File: cmd/helm/testdata/testcharts/alpine/templates/alpine-pod.yaml (hunk 1 above)
Comment (luisdavim) [reply cmd/helm/testdata/testcharts/alpine/templates/alpine-pod.yaml]: I've added `appVersion` to the `Chart.yaml`

**Comment by luisdavim** [comment]
Thanks for the review @mattfarina I think I've addressed all your comments.

**Comment by mattfarina** [comment]
@luisdavim Can you please fix the tests?

**Comment by luisdavim** [comment]
sure, I'll try to get that done by the end of the week.

**Comment by luisdavim** [comment]
it was easier than I thought, they're fixed now.

File: docs/chart_template_guide/variables.md [hunk 6]
```diff
@@ -114,6 +114,8 @@ metadata:
     # I cannot reference .Chart.Name, but I can do $.Chart.Name
     helm.sh/chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
     app.kubernetes.io/instance: "{{ $.Release.Name }}"
+    # Value from appVersion in Chart.yaml
+    app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
```
Comment (mattfarina) [inline docs/chart_template_guide/variables.md]: This is a nit but the other docs in this section start with `$.Chart...`. This instance should hold the pattern to be consistent. Can you please update it.

**Review by mattfarina — COMMENTED** [review]
Thanks for making the changes. I have just one more nit for doc consistency...

File: docs/chart_template_guide/variables.md (hunk 6 above)
Comment (luisdavim) [reply docs/chart_template_guide/variables.md]: updated.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**